x-team/xp-ui
# Auto build XP UI upon installing a package

**Release Type:** *Dev Improvement* 


## Description

**THIS ISN'T A BREAKING CHANGE. THE APP SHOULD STILL WORK AS BEFORE**

Currently, we're forced to build `xp-ui` each time we're about to open a PR or before we merge the PR. This is error-prone and repetitive. There's no even a need to keep `/lib` files inside the repository and commit it to the repo. Moreover, currently, when we install `xp-ui` lib into consumer client apps we carry all the content of `xp-ui` repository. All we need is simply a `lib/` folder with built files and some essential files like `README.md` and `LICENSE`. 

The automation we perform here is by utilizing NPM's `prepare` script which ensures to run commands specified (`npm run build` in our case) when installing `xp-ui` into the client app.

Apart from that, I've performed some cleanup of redundant things like `zappr config file`, and some unused webpack config fields.

## Checklist

**Before submitting a Pull Request,** please make sure the following is done:



## Steps to Test or Reproduce

To test it out locally, to verify `xp-registration` runs correctly, proceed with the following steps.

1. Go into `xp-registration` in your terminal (use `develop` branch)
1. Remove installed `xp-ui` with `npm un xp-ui -S`
1. Install `xp-ui` from the current branch by running `npm i git://github.com/x-team/xp-ui.git#auto-build`
1. Start the `xp-reg` app with `npm start` (make sure you have `xp-api` running) and check the app works as before.
1. Ensure `npm run build` passes successfully. 
1. Once tested, ensure to revert installed `xp-ui` to the `develop` one if you're testing on `develop` branch.

## Impacted Areas in Application

The `xp-ui` build flow is updated.

## Screenshots

#### The content of installed `xp-ui` package

![image](https://user-images.githubusercontent.com/579331/81062003-f5bde180-8edd-11ea-95b4-14b883350cd4.png)

#### The lib folder content

![image](https://user-images.githubusercontent.com/579331/81064352-391a4f00-8ee2-11ea-9e46-a083a3adff55.png)
